Redesign: fix eye-icon on password fields when there's an error

greenwoodt opened the pull request:
<!--
NOTE: We are in the middle of a big redesign of the frontend.
That could mean that your PR will not get through on the next few months.
Please see https://github.com/decidim/decidim/discussions/9512
-->

#### :tophat: What? Why?
The eye-icon which hides your password in the form when confirming your password is incorrect was not adjusted properly. A change to ```bottom``` element in button allows for the icon to be centered properly.

#### :pushpin: Related Issues
*Link your PR to an issue*
- Fixes #11789 

#### Testing
1. Login 
2. Head over to account
3. Find change your password
4. In the fields asking to confirm your "new" and "current", make sure they're both different from each other.
5. Click change
6. See icon stay centred in the field box.

### :camera: Screenshots

<img width="801" alt="Screenshot 2023-10-23 at 14 26 14" src="https://github.com/decidim/decidim/assets/101816158/6bac890d-c9f9-4790-8ed6-ce9daa88302d">


:hearts: Thank you!

alecslupu reviewed (dismissed):
LGTM

greenwoodt commented:
> LGTM

I'll have to revisit this. Seems this change is breaking other module tests.

andreslucena commented:
@greenwoodt seeing the classes names, I say that the problem comes from my implementation of this feature. I tried to not change that much of the CSS and reuse the `filter-search` when we should actually have something like `input-group` that's used in these kind of inputs with icons. 

I'd also change the `filter-container` class to `input-group-container` too.

As the password toggler is implemented in JS, this is done here 

https://github.com/decidim/decidim/blob/f1643836e7522cfc0ffab0c6f5adc6855a6f2958/decidim-core/app/packs/src/decidim/password_toggler.js#L83

andreslucena commented:
@alecslupu I wanted to cancel your review, just in case we merge this wrongly. Marking this as draft until @greenwoodt tell us that this is ready again

greenwoodt commented:
I have updated line 20 in ```decidim-core/app/packs/src/decidim/password_toggler.js``` to adjust the height of the class when its invalid. An issue (unconfirmed bug) occurs when you adjust the height to be same as when its ```OnValid: "mt-5"```, it makes the eye-icon seem like it has quite high margin. 

I have also changed the class names to see if this has an effect with the test issues.

andreslucena reviewed (dismissed):
I've just checked it out and LGTM

There's a small change in the button displacement, but I think I found it because I was looking for it

alecslupu reviewed (changes requested):
@greenwoodt can you have a look on the failing issues ( a core spec and some linting )

andreslucena reviewed (approved):
I just give it a last look and I'm much more happy with the removal of the abide hacks. It seems much more clean IMO 👍🏽 

I'll leave the merge to @alecslupu as he gave the last review

alecslupu reviewed (approved):
:+1:
